framework/src/materials/Material.C

```diff
@@ -171,8 +175,26 @@ Material::getOutputs()
 void
 Material::computeProperties()
 {
-  for (_qp = 0; _qp < _qrule->n_points(); ++_qp)
+  // If this Material has the _constant_on_elem flag set, we take the
+  // value computed for _qp==0 and use it at all the quadrature points
+  // in the Elem.
+  if (_constant_on_elem)
+  {
+    // Compute MaterialProperty values at the first qp.
+    _qp = 0;
     computeQpProperties();
+
+    // Copy the first value computed over to the others.
+    MaterialProperties & props = _material_data->props();
```

Comment:
So for every property in `_material_data->props()`, should I check whether it is in the `_supplied_props` for the current `Material` before deciding whether to do the copy? Or is there a better way to ask a `Material` what properties have been declared on it?